task/runner.go

```diff
@@ -0,0 +1,101 @@
+package task
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/google/uuid"
+	"github.com/turbot/steampipe/constants"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+type Runner struct {
+	currentState state
+	shouldRun    bool
+}
+
+func NewRunner() *Runner {
+	r := new(Runner)
+	r.loadState()
+	r.shouldRun = r.getShouldRun()
+	return r
+}
+
+func (r *Runner) Run() {
+	if r.shouldRun {
+		checkVersion(r.currentState.InstallationID)
+		// update last check time
+		r.updateState()
+	}
+}
+
+func (r *Runner) loadState() {
+	stateFilePath := filepath.Join(constants.InternalDir(), updateStateFileName)
+	// get the state file
+	_, err := os.Stat(stateFilePath)
+	if err != nil {
+		r.currentState = r.createState()
+		return
+		// create folder structure if not there
+	}
+
+	stateFileContent, err := ioutil.ReadFile(stateFilePath)
+	if err != nil {
+		fmt.Println("Could not read update state file")
+		r.currentState = r.createState()
+		return
+	}
+
+	err = json.Unmarshal(stateFileContent, &r.currentState)
+	if err != nil {
+		fmt.Println("Could not parse update state file")
+		r.currentState = r.createState()
+		return
+	}
+}
+
+func (r *Runner) getShouldRun() bool {
+	if v, ok := os.LookupEnv(disableEnvVarName); ok && v == "true" {
```

Comment:
this check is _only_ to disable the update check, not all task running - move to the start of checkVersion